### ISSUE TITLE
Selecting tether shooting with a Fuji X-H1 plugged (with camera in tether mode) crashes Darktable

### ISSUE THREAD

**Issue by LucasGGamerM** [darktable-org/darktable#17018]
### Describe the bug

Trying to activate a tether connection makes the program crash. 

### Steps to reproduce

Plug camera into computer (with the camera being in tether mode), open Darktable, unmount the camera from the filesystem, click on the start tethering button, and see the crash. 

### Expected behavior

Darktable should start a tethering connection. 

### Logfile | Screenshot | Screencast

I couldn't find them on the flatpak, but I am willing to give it once I find the backtrace.

### Where did you obtain darktable from?

flatpak

### darktable version

4.6.1

### What OS are you using?

Linux

### What is the version of your OS?

PopOS 22.04

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

Nvidia 3050, using latest available driver

**Comment by paperdigits** [comment]
Have you made sure that the flatpak has the proper permissions to access the tethered camera?

**Comment by LucasGGamerM** [comment]
> Have you made sure that the flatpak has the proper permissions to access the tethered camera?

I am not sure. How can I check that?

**Comment by paperdigits** [comment]
Get the application "Flatseal" its an easy GUI to inspect the sandbox permissions of a flatpak.

**Comment by LucasGGamerM** [comment]
> Get the application "Flatseal" its an easy GUI to inspect the sandbox permissions of a flatpak.

It has access to all devices, and copy and import works. But not tethered shooting

**Comment by paperdigits** [comment]
Is the camera storage mounted by your desktop environment automatically when it is connected?

Also is your camera supported by libgphoto2?

**Comment by LucasGGamerM** [comment]
> Is the camera storage mounted by your desktop environment automatically when it is connected? Also is your camera supported by libgphoto2?
> […](#)
> On June 21, 2024 4:30:59 PM PDT, LucasGGamerM ***@***.***> wrote: > Get the application "Flatseal" its an easy GUI to inspect the sandbox permissions of a flatpak. It has access to all devices, and copy and import works. But not tethered shooting -- Reply to this email directly or view it on GitHub: [#17018 (comment)](https://github.com/darktable-org/darktable/issues/17018#issuecomment-2183579246) You are receiving this because you commented. Message ID: ***@***.***>

Yes, as per https://github.com/gphoto/libgphoto2/wiki/List-of-cameras it is.

**Comment by paperdigits** [comment]
Alright, how about the first question?

**Comment by LucasGGamerM** [comment]
> Alright, how about the first question?
> […](#)
> On June 21, 2024 4:59:14 PM PDT, LucasGGamerM ***@***.***> wrote: > Is the camera storage mounted by your desktop environment automatically when it is connected? Also is your camera supported by libgphoto2? > […](#) > On June 21, 2024 4:30:59 PM PDT, LucasGGamerM ***@***.***> wrote: > Get the application "Flatseal" its an easy GUI to inspect the sandbox permissions of a flatpak. It has access to all devices, and copy and import works. But not tethered shooting -- Reply to this email directly or view it on GitHub: [#17018 (comment)]([#17018 (comment)](https://github.com/darktable-org/darktable/issues/17018#issuecomment-2183579246)) You are receiving this because you commented. Message ID: ***@***.***> Yes, as per https://github.com/gphoto/libgphoto2/wiki/List-of-cameras it is. -- Reply to this email directly or view it on GitHub: [#17018 (comment)](https://github.com/darktable-org/darktable/issues/17018#issuecomment-2183592116) You are receiving this because you commented. Message ID: ***@***.***>

Yes.

**Comment by paperdigits** [comment]
Can you unmount it and forbid your desktop from mounting it?

Secondly, if you install gphoto2 and try using it from the command line, does it work?

**Comment by LucasGGamerM** [comment]
I can unmount it, but I am not sure how to forbid it. 

And yes, yesterday I was messing around with gphoto2 in the command line and the live view feature and taking a picture directly from the command line worked fine.

**Comment by robyquin** [comment]
Same thing with Nikon D7500+DarkTable 4.8.1 (windows10)

I attach debug log (**darktable -d camctl**)
[log_20240906_1.txt](https://github.com/user-attachments/files/16911749/log_20240906_1.txt)

**NOTE**: if I use **gphoto2** directly, I can capture images without problems

**Comment by robyquin** [comment]
I compiled last version from github, and this version also crashes!
(Nikon D7500+darktable 4.9.0+421~gdb0990bfed - windows10)

[log_20240906_4.9.0+421~gdb0990bfed.txt](https://github.com/user-attachments/files/16912160/log_20240906_4.9.0_%2B421.gdb0990bfed.txt)

**Comment by paperdigits** [comment]
@robyquin you need to have your DE not mount it, this is from your log: 

>    36,3377 [camera_control] remove Nikon DSC D7500 on port usb:001,011 from ununsed camera list as mounted

**Comment by robyquin** [comment]
This weekend I did some tests with different settings, Here are log files log:

[log_20240906_4.9.0+421~gdb0990bfed_magik.txt](https://github.com/user-attachments/files/16926992\log_20240906_4.9.0%2B421.gdb0990bfed_magik.txt)
[log_20240906_4.9.0+421~gdb0990bfed_noCL.txt](https://github.com/user-attachments/files/16926993/log_20240906_4.9.0%2B421.gdb0990bfed_noCL.txt)
[log_20240906_4.9.0+421~gdb0990bfed_OutofFocus.txt](https://github.com/user-attachments/files/16926994/log_20240906_4.9.0%2B421.gdb0990bfed_OutofFocus.txt)
[log_20240906_4.9.0+421~gdb0990bfed-dirty.txt](https://github.com/user-attachments/files/16926995/log_20240906_4.9.0%2B421.gdb0990bfed-dirty.txt)
[log_20240906_4.9.0+421~gdb0990bfed.txt](https://github.com/user-attachments/files/16926997/log_20240906_4.9.0%2B421.gdb0990bfed.txt)

in this log, I added new log messages to understand the flow of function calls:
[log_20240907_4.9.0+421~gdb0990bfed-dirty_robyquin-log.txt](https://github.com/user-attachments/files/16926996/log_20240907_4.9.0%2B421.gdb0990bfed-dirty_robyquin-log.txt)

**An interesting thing, I hope to help you**

I had to finish a photographic work, and as a workaround, I used "copy & import from camera" button: It works fine!
(I have no log files of this last attempt.)
